src/components/Puzzle.tsx

```diff
@@ -1,20 +1,47 @@
 import '../css/Puzzle.css';
-// import PuzzleDetails from '../components/PuzzleDetails'
+import Modal from 'react-modal';
+import { useState } from 'react'
+import React from 'react'
+import PuzzleDetails from './PuzzleDetails'
+
+Modal.setAppElement('#root');
 
 interface puzzleProps {
   id: string
   pieceCount: string
   image: string
+  category: string
+  missingPieces: string
+  price: string 
+  quality: string
 }
 
-const Puzzle: React.FC<puzzleProps> = ({ id, pieceCount, image }) => {
+const Puzzle: React.FC<puzzleProps> = ({ id, pieceCount, image, category, quality, missingPieces, price }) => {
+  const [modalOpen, setModalOpen] = useState(false)
+
+  const openModal = (event: React.MouseEvent) => {
+    event.preventDefault()
+    setModalOpen(true)
+  }
+
+  const closeModal = (event: React.MouseEvent) => {
+    event.preventDefault()
+    setModalOpen(false)
+  }
 
   return (
     <section>
-      <div className='individual-puzzle' onClick={() => console.log(id)}>
+      <div className='individual-puzzle' onClick={event => openModal(event)}>
         <img src={image} />
       </div>
       <h4 className='piece-count'>{pieceCount} pieces</h4>
+      <Modal 
```

Comment:
We should add a property in the modal component to stop the scrolling in the background. We should also consider aria labels in the modal for screen readers.